### PR TITLE
Fix/csi 1545 powerpc fc subsystem

### PR DESCRIPTION
**Pull request by oriyarde**
change PowerPC FC `/dev/disk/by-path` search from:
**`pci-*`**`fc-0x0000000000000000-lun-0 -> ../../sdX`
to:
`fc-0x0000000000000000-lun-0 -> ../../sdX` (without `pci-*` prefix)

### CODE REVIEW

File: node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go [hunk 1]
```diff
@@ -226,7 +228,7 @@ func (r OsDeviceConnectivityHelperScsiGeneric) GetMpathDevice(volumeId string, l
 func (r OsDeviceConnectivityHelperScsiGeneric) waitForMpath(targetPath string, connectivityType string, arrayIdentifier string, lunId int, volumeId string,
 	resChannel chan<- *WaitForMpathResult) {
 	lunIdStr := convertIntToScsilun(lunId)
-	dp := strings.Join([]string{targetPath, connectivityType, arrayIdentifier, "lun", lunIdStr}, "-")
+	dp := targetPath + strings.Join([]string{connectivityType, arrayIdentifier, "lun", lunIdStr}, "-")
```
Comment (igorgonibm) [inline node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go:231]: why this change?

File: node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go [hunk 2]
```diff
@@ -155,20 +155,22 @@ func (r OsDeviceConnectivityHelperScsiGeneric) GetMpathDevice(volumeId string, l
...
 		// In host, the path like this: /dev/disk/by-path/pci-0000:13:00.0-fc-0x500507680b25c0aa-lun-0
 		// So add prefix "0x" for the arrayIdentifiers
 		for index, wwn := range arrayIdentifiers {
 			arrayIdentifiers[index] = "0x" + strings.ToLower(wwn)
 		}
 	}
 	if connectivityType == ConnectionTypeISCSI {
-		targetPath = "/dev/disk/by-path/ip*"
+		subsystemPrefix = "ip*-"
 	}
```
Comment (igorgonibm) [inline node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go:171]: leave devicePath instance variable as before and init it here. Don't define it here as local variable.
What is the reason u devicePath as instance variable?

File: node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go (hunk 1 above)
Comment (oriyarde) [reply node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go:231]: we don't want the hyphen in case there is no prefix before `fc-0x0000000000000000-lun-0`

File: node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go (hunk 2 above)
Comment (oriyarde) [reply node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go:171]: I'm not sure I understand what you mean.
could you clarify?

Comment (igorgonibm) [reply node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go:171]: I mean targetPath
Leave it as instance variable
and don't add subsystemPrefix  as instance variable

Comment (oriyarde) [reply node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go:171]: why?
and are you implying `targetPath` is not an instance variable in my current change? if so, why?

Comment (igorgonibm) [reply node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go:171]: My mistake sorry. They are not instance variables.

File: node/pkg/driver/device_connectivity/device_path_amd64.go [hunk 3]
```diff
@@ -0,0 +1,4 @@
+package device_connectivity
+
+// for x86, the FC subsystem prefix is pci
```
Comment (igorgonibm) [inline node/pkg/driver/device_connectivity/device_path_amd64.go:3]: I like comments but here it just says exactly as the code line itself

File: node/pkg/driver/device_connectivity/device_path_ppc64le.go [hunk 4]
```diff
@@ -0,0 +1,4 @@
+package device_connectivity
+
+// for PowerPC, no FC subsystem prefix is needed
```
Comment (igorgonibm) [inline node/pkg/driver/device_connectivity/device_path_ppc64le.go:3]: better change to "for Power PC there is no FC subsystem prefix"

File: node/pkg/driver/device_connectivity/device_path_s390x.go [hunk 5]
```diff
@@ -1,4 +1,4 @@
 package device_connectivity
 
 // on Z-Systems, the udev subsystem is a Channel Command Word
-const fcSubsystem = "ccw"
```
Comment (igorgonibm) [inline node/pkg/driver/device_connectivity/device_path_s390x.go:4]: also here comment is not needed

File: node/pkg/driver/device_connectivity/device_path_ppc64le.go (hunk 4 above)
Comment (oriyarde) [reply node/pkg/driver/device_connectivity/device_path_ppc64le.go:3]: do you know this for a fact? because I don't

File: node/pkg/driver/device_connectivity/device_path_s390x.go (hunk 5 above)
Comment (oriyarde) [reply node/pkg/driver/device_connectivity/device_path_s390x.go:4]: the comment is not part of this PR (and you approved it in https://github.com/IBM/ibm-block-csi-driver/pull/124).
then why is it needed in Power?

File: node/pkg/driver/device_connectivity/device_path_amd64.go (hunk 3 above)
Comment (oriyarde) [reply node/pkg/driver/device_connectivity/device_path_amd64.go:3]: then why do we want to keep the comment in Power?

File: node/pkg/driver/device_connectivity/device_path_ppc64le.go (hunk 4 above)
Comment (igorgonibm) [reply node/pkg/driver/device_connectivity/device_path_ppc64le.go:3]: This is what you did here. Prefix is empty string -> no prefix.

File: node/pkg/driver/device_connectivity/device_path_amd64.go (hunk 3 above)
Comment (igorgonibm) [reply node/pkg/driver/device_connectivity/device_path_amd64.go:3]: I would remove it in power too.

File: node/pkg/driver/device_connectivity/device_path_s390x.go (hunk 5 above)
Comment (igorgonibm) [reply node/pkg/driver/device_connectivity/device_path_s390x.go:4]: I would remove it in power too.

File: node/pkg/driver/device_connectivity/device_path_ppc64le.go (hunk 4 above)
Comment (oriyarde) [reply node/pkg/driver/device_connectivity/device_path_ppc64le.go:3]: doesn't mean that I know that FC **never** (not even outside CSI context) has a prefix on PowerPC

Comment (oriyarde) [reply node/pkg/driver/device_connectivity/device_path_ppc64le.go:3]: as I understood from the other conversations, I'm removing all 3 comments

File: node/pkg/driver/device_connectivity/device_path_amd64.go (hunk 3 above)
Comment (oriyarde) [reply node/pkg/driver/device_connectivity/device_path_amd64.go]: done

File: node/pkg/driver/device_connectivity/device_path_s390x.go (hunk 5 above)
Comment (oriyarde) [reply node/pkg/driver/device_connectivity/device_path_s390x.go:4]: done
